sqlmesh/core/context.py

```diff
@@ -2485,6 +2457,19 @@ def lint_models(
                 "Linter detected errors in the code. Please fix them before proceeding."
             )
 
+    def load_model_tests(
+        self, tests: t.Optional[t.List[str]] = None, patterns: list[str] | None = None
+    ) -> t.List[ModelTestMetadata]:
+        # If a set of tests is provided, use a single loader to load them
+        # Otherwise, gather all tests from all loaders/repos
+        loaders = [self._loaders[0]] if tests else self._loaders
```

Comment:
We could improve the comment here, explaining why any old loader does the trick if `tests` is provided. The reason is that the test path is part of the argument, so there's no need to walk every `tests/` folder in each repo, and we just need an arbitrary loader only for the loading logic.